src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleKeyCache.scala

```diff
@@ -41,59 +47,65 @@ class GoogleKeyCache(
         this
       ))
 
-    googleStorageDAO.createBucket(googleServicesConfig.serviceAccountClientProject, googleServicesConfig.googleKeyCacheConfig.bucketName).recover {
-      case t: GoogleJsonResponseException if t.getDetails.getMessage.contains("You already own this bucket") && t.getDetails.getCode == 409 => ()
+    googleStorageAlg.createBucket(googleServicesConfig.serviceAccountClientProject, googleServicesConfig.googleKeyCacheConfig.bucketName, List.empty).recoverWith {
+      case t: StorageException if t.getCode == 409 && t.getMessage.contains("You already own this bucket") =>
+        IO(logger.info(t.getMessage))
     } flatMap { _ =>
-      googleStorageDAO.setBucketLifecycle(googleServicesConfig.googleKeyCacheConfig.bucketName, googleServicesConfig.googleKeyCacheConfig.retiredKeyMaxAge)
+      val lifecycleCondition = BucketInfo.LifecycleRule.LifecycleCondition.newBuilder()
+                                          .setAge(googleServicesConfig.googleKeyCacheConfig.retiredKeyMaxAge)
+                                          .build()
+      val lifecycleRule = new LifecycleRule(LifecycleRule.LifecycleAction.newDeleteAction(), lifecycleCondition)
+      googleStorageAlg.setBucketLifecycle(googleServicesConfig.googleKeyCacheConfig.bucketName, List(lifecycleRule))
     }
   }
 
-  override def getKey(pet: PetServiceAccount): Future[String] = {
-    val retrievedKeys = for {
-      keyObjects <- googleStorageDAO.listObjectsWithPrefix(
-        googleServicesConfig.googleKeyCacheConfig.bucketName,
-        keyNamePrefix(pet.id.project, pet.serviceAccount.email))
-      keys <- googleIamDAO.listUserManagedServiceAccountKeys(pet.id.project, pet.serviceAccount.email).map(_.toList)
-      cleanedKeyObjects <- cleanupUnknownKeys(pet, keyObjects, keys)
-    } yield (cleanedKeyObjects, keys)
-
-    retrievedKeys.flatMap {
-      case (Nil, _) => furnishNewKey(pet) //mismatch. there were no keys found in the bucket, but there may be keys on the service account
-      case (_, Nil) => furnishNewKey(pet) //mismatch. there were no keys found on the service account, but there may be keys in the bucket
-      case (keyObjects, serviceAccountKeys) => retrieveActiveKey(pet, keyObjects, serviceAccountKeys)
-    }
-  }
+  override def getKey(pet: PetServiceAccount): IO[String] =
+    for {
+      keyObjects <- googleStorageAlg
+        .unsafeListObjectsWithPrefix(googleServicesConfig.googleKeyCacheConfig.bucketName, keyNamePrefix(pet.id.project, pet.serviceAccount.email))
+      keys <- IO.fromFuture(IO(googleIamDAO.listUserManagedServiceAccountKeys(pet.id.project, pet.serviceAccount.email).map(_.toList)))
+      cleanedKeyObjects <- IO.fromFuture(IO(cleanupUnknownKeys(pet, keyObjects, keys)))
+      res <- (cleanedKeyObjects, keys) match {
```

Comment:
avoid abbreviations: `result` instead of `res`

Reply:
isn't `res` common enough to abbreviate?

Reply:
no, my first thought is that it means `resource`

There are trade offs with abbreviations. They usually make code harder to understand. They can make it easier but only when names get too long. They make the code easier to write but that is pretty much completely offset by autocomplete.

So in the case of `res`, what it gained in exchange for decreased readability?

Reply:
fwiw, I never abbreviated `resource` as `res` and I agree abbreviating `resource` as `res` would be really confusing...I thought `res` or sometimes `ret` is common enough that there's no confusion at all